math/src/field/f64/mod.rs

```diff
@@ -247,7 +247,11 @@ impl StarkField for BaseElement {
         let x = self.0;
         let (r, c) = x.overflowing_add(x << 32);
         let res = r.wrapping_sub(r >> 32).wrapping_sub(c as u64);
-        M - res
+        let res = M - res;
+
+        // Constant-time branching, handling 0 case.
+        let mask = -((self.0 == 0) as i64) as u64;
+        res ^ (mask & res)
```

Comment:
This works! But I'm wondering if this is better than:
```Rust
let x = self.0;
let (a, e) = x.overflowing_add(x << 32);
let b = a.wrapping_sub(a >> 32).wrapping_sub(e as u64);

let (r, c) = 0u64.overflowing_sub(b);
r.wrapping_sub(0u32.wrapping_sub(c as u32) as u64)
```

On [godbolt](https://godbolt.org/) the above seems to give a very marginally better result:

![Screenshot 2023-02-18 at 12 43 06 PM](https://user-images.githubusercontent.com/68976082/219898629-1b3b1ba7-ace5-4e9a-8bca-5f3b2cb05f3d.png)

vs.

![Screenshot 2023-02-18 at 12 44 14 PM](https://user-images.githubusercontent.com/68976082/219898757-8ce6876f-c885-4d2e-87bc-1fb52b815d5c.png)

Reply:
Yeah - let's do it. Thank you!

Reply:
Fixed, sorry about that!